src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -839,10 +859,24 @@ export class ShelleyWallet extends Wallet implements WalletInterface {
   }
 
   async fetchUTXOs() {
-    return api.bulkFetchUTXOsForAddresses(
-      [...this.internalAddresses, ...this.externalAddresses],
-      this._getBackendConfig(),
-    )
+    if (!this.utxoService) throw new Error('utxoService not initialized')
+
+    const addresses = this.externalAddresses.concat(this.internalAddresses)
+    await this.utxoService.syncUtxoState(addresses)
+    const utxos = await this.utxoService.getAvailableUtxos()
+
+    return utxos.map((utxo) => ({
+      utxo_id: utxo.utxoId,
+      tx_hash: utxo.txHash,
+      tx_index: utxo.txIndex,
+      amount: utxo.amount,
+      receiver: utxo.receiver,
+      assets: utxo.assets,
+    })) as unknown as RawUtxo[]
+  }
```

Comment:
can we have a helper function `toRawUtxo` ?

Reply:
can we get rid of `Raw`?
what does that even mean?

Reply:
it's legacy type :/ totally agree.

Reply:
Done thanks